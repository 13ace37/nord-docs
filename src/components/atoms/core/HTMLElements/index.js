/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

/**
 * @file Provides components that represents basic HTML elements.
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element
 * @since 0.3.0
 */

import { A } from "./inlineTextSemantics";

/* eslint-disable-next-line import/prefer-default-export */
export { A };
