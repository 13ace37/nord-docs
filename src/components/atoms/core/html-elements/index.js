/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

/**
 * @file Provides components that represent basic HTML elements.
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @since 0.3.0
 * @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element
 */

import Input from "./forms";
import { A, Code } from "./inline-text-semantics";
import { Details, Summary } from "./interactive";
import { H1, H2, H3, H4, H5, H6 } from "./sectioning";
import { Caption, Table, Td, Th, Tr } from "./tabular";
import { Blockquote, Figure, Hr, Li, Ol, P, Pre, Ul } from "./text";

export {
  A,
  Blockquote,
  Caption,
  Code,
  Details,
  Figure,
  H1,
  H2,
  H3,
  H4,
  H5,
  H6,
  Hr,
  Input,
  Li,
  Ol,
  P,
  Pre,
  Summary,
  Table,
  Td,
  Th,
  Tr,
  Ul
};
