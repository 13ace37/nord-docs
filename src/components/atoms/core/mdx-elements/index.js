/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import Image from "./Image";
import ShrinkedWidth from "./ShrinkedWidth";
import Video from "./Video";

export { Image, ShrinkedWidth, Video };
