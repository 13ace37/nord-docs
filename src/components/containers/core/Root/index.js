/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import Root, { GlobalThemeMode } from "./Root";

export { GlobalThemeMode };
export default Root;
