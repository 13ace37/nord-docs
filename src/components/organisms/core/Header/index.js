/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import { HEADER_BRAND_LOGO_SHRINK_FACTOR, HEADER_HEIGHT, HEADER_HEIGHT_PINNED } from "./shared/styles";
import Header from "./Header";

export { HEADER_BRAND_LOGO_SHRINK_FACTOR, HEADER_HEIGHT, HEADER_HEIGHT_PINNED };
export default Header;
