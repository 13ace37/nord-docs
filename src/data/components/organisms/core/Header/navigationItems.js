/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

/**
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @since 0.3.0
 */

import { ROUTE_BLOG, ROUTE_COMMUNITY, ROUTE_DOCS, ROUTE_PORTS } from "config/routes/mappings";

/**
 * The mapping of navigation link item names to their target URL.
 *
 * @since 0.3.0
 */
const navigationItems = [
  { title: "Ports", url: ROUTE_PORTS },
  { title: "Docs", url: ROUTE_DOCS },
  { title: "Blog", url: ROUTE_BLOG },
  { title: "Community", url: ROUTE_COMMUNITY }
];

export default navigationItems;
