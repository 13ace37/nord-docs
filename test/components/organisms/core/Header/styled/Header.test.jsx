/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import React from "react";
import { em } from "polished";

import { renderWithTheme } from "nord-docs-test-utils";
import Header from "organisms/core/Header/styled/Header";
import { HEADER_HEIGHT, HEADER_HEIGHT_PINNED } from "organisms/core/Header";

describe("theme styles", () => {
  test("uses flexbox layout", () => {
    const { container } = renderWithTheme(<Header />);
    expect(container.firstChild).toHaveStyleRule("display", "flex");
  });

  test("has fixed position", () => {
    const { container } = renderWithTheme(<Header />);
    expect(container.firstChild).toHaveStyleRule("position", "fixed");
  });

  test("has correct height in unpinned mode", () => {
    const { container } = renderWithTheme(
      <Header height={HEADER_HEIGHT} heightPinned={HEADER_HEIGHT_PINNED} isPinned={false} />
    );

    expect(container.firstChild).toHaveStyleRule("height", em(HEADER_HEIGHT));
    expect(container.firstChild).toMatchSnapshot();
  });

  test("has correct height in pinned mode", () => {
    const { container } = renderWithTheme(
      <Header height={HEADER_HEIGHT} heightPinned={HEADER_HEIGHT_PINNED} isPinned />
    );

    expect(container.firstChild).toHaveStyleRule("height", em(HEADER_HEIGHT_PINNED));
    expect(container.firstChild).toMatchSnapshot();
  });
});
