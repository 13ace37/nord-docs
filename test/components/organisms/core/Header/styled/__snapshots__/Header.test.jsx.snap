// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`theme styles has correct height in pinned mode 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  top: 0;
  height: 3.5em;
  width: 100%;
  z-index: 3;
  background-color: #fff;
  box-shadow: 0 5px 10px 0 rgba(216,222,233,0.15);
  -webkit-transition: height 250ms ease-in-out, box-shadow 250ms ease-in-out, background-color 400ms ease-in-out;
  transition: height 250ms ease-in-out, box-shadow 250ms ease-in-out, background-color 400ms ease-in-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<header
  class="c0"
  height="80"
/>
`;

exports[`theme styles has correct height in unpinned mode 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  top: 0;
  height: 5em;
  width: 100%;
  z-index: 3;
  background-color: #fff;
  -webkit-transition: height 250ms ease-in-out, box-shadow 250ms ease-in-out, background-color 400ms ease-in-out;
  transition: height 250ms ease-in-out, box-shadow 250ms ease-in-out, background-color 400ms ease-in-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<header
  class="c0"
  height="80"
/>
`;
